### PR TITLE
STM32: add support for STM32G031 SoC and STM32G0316-DISCO development board

### PR DESCRIPTION
**Pull request by fbrozovic**

### CODE REVIEW

**Review by erwango — CHANGES_REQUESTED** [review]
8 pins, 8 kb of RAM... interesting challenge!
Few comments

File: soc/arm/st_stm32/stm32g0/Kconfig.defconfig.stm32g031j6 [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+if SOC_STM32G031XX
+
+config SOC
+	string
+	default "stm32g031xx"
+
+config NUM_IRQS
+	int
+	default 32
```
[the remark this answers is not in the document]
Comment (fbrozovic) [reply soc/arm/st_stm32/stm32g0/Kconfig.defconfig.stm32g031j6]: OK, I was only looking at the vector table in the reference manual and didn't see that G031 doesn't have the CEC and AES/RNG peripherals.

**Comment by fbrozovic** [comment]
@erwango Comments addressed.
Any idea how to fix the test that fails to compile due to the too small SRAM? I was thinking of adding min_ram: 12 to `tests/kernel/pipe/pipe_api/testcase.yaml`, but I'm not sure about best practices here...

**Comment by erwango** [comment]
@fbrozovic 
> I was thinking of adding min_ram: 12 to tests/kernel/pipe/pipe_api/testcase.yaml, but I'm not sure about best practices here...

First thing would be to set a minimal configuration first, like disabling the GPIO ports you don't use for basics such as console, button, led.
Let's see what happens then.
Indeed, last step would be to set min_ram on tests, but I'm not comfortable to do that too much widely.

**Comment by fbrozovic** [comment]
@erwango I completely forgot about the possibility of lowering the default stack sizes - I've done that now, hopefully it'll be enough to build the tests.

**Review by dbkinder — APPROVED** [review]
doc changes LGTM, thanks.

**Review by erwango — CHANGES_REQUESTED** [review]
@fbrozovic , looks good.
Can you take care of the issue reported by checkpatch (a part from UNDOCUMENTED_DT_STRING) ?

**Comment by fbrozovic** [comment]
@erwango I think I can only fix the SPACING warnings, but not the MISSING_BREAK and LONG_LINE ones?

**Comment by erwango** [comment]
> @erwango I think I can only fix the SPACING warnings, but not the MISSING_BREAK and LONG_LINE ones?

@fbrozovic, that is fine, txs.

**Review by erwango — APPROVED** [review]

**Pull request merged**